The mozilla startup script searches for the run_mozilla.sh script in
various places before defaulting to the mozilla libdir. Once it settles
for the mozilla libdir it should record the path to the run_mozilla.sh
directory as well...

--- mozilla/browser/app/mozilla.in.bad	2005-10-12 11:44:22.000000000 -0400
+++ mozilla/browser/app/mozilla.in	2005-10-12 11:45:54.000000000 -0400
@@ -110,6 +110,7 @@
   # Check default compile-time libdir
   if [ -x "$moz_libdir/run-mozilla.sh" ]; then
     dist_bin="$moz_libdir"
+    run_moz="$dist_bin/run-mozilla.sh"
   else 
     echo "Cannot find mozilla runtime directory. Exiting."
     exit 1
